src/main/java/com/devonfw/devcon/modules/oasp4j/Oasp4j.java

```diff
@@ -571,4 +570,27 @@ private Node addNode(Document doc, String packagname) {
     return dependency;
   }
 
+  /**
+   * Gets the template version from the version.json file. Firstly, it's searching for it on devonfw.github.io
+   * repository. If not found, it's searching for it in the DevonFolder/conf/ folder.
+   *
+   * @param serverPath Path where DEVON server is located on disk
+   * @return The template version or empty string if not found
+   */
+  private String getTemplateVersion(String serverPath) {
+
+    String oaspTemplateVersion = "";
+    Optional<String> oaspTemplateVersion_op = Downloader.getDevconConfigProperty(Constants.OASP_TEMPLATE_VERSION); // Optional.of("2.3.0");
+    if (oaspTemplateVersion_op.isPresent()) {
+      oaspTemplateVersion = oaspTemplateVersion_op.get();
+    } else {
+      String configPath = Utils.addTrailingSlash(serverPath) + Constants.VERSION_PARAMS_FILE_FULL_PATH;
```

Comment:
We should implement it in the way, that the configuration takes precedence over all other sources.
Currently, the main issue was that the behavior differs if you are only vs. you are offline. This should not be the case as this causes issues in semi-offline scenarios like trainings.

Reply:
OK! I will change it as you mention: the template version will only be read from local file "version.json"

Reply:
@maybeec Just confirmation over here that did you mean removing below line 
Optional<String> oaspTemplateVersion_op = Downloader.getDevconConfigProperty(Constants.OASP_TEMPLATE_VERSION);

So now it is only taking the version.json from DevonServerPath/conf whether the network is offline or online . It no more considers the version.json from github. 

I have tested this scenario it throws below error if I dont have version.json in DevonServerPath/conf with network offline/online
Oasp template version not found in config file.